From 15cd9b830f0bfcba4b3676905c075090bdead337 Mon Sep 17 00:00:00 2001
From: Shwetha G K <quic_kshwetha@quicinc.com>
Date: Mon, 5 Feb 2024 15:39:31 +0530
Subject: [PATCH] cfg80211: Remove mld_iface_name support

Remove mld_iface_name from vifs params and cleanup
its usage

Change-Id: Idbf026172f40521e44591a9fb4f35669f25e4ef9
Signed-off-by: Shwetha G K <quic_kshwetha@quicinc.com>
---
 include/net/cfg80211.h       | 2 --
 include/uapi/linux/nl80211.h | 1 -
 net/wireless/nl80211.c       | 9 ---------
 3 files changed, 12 deletions(-)

diff --git a/include/net/cfg80211.h b/include/net/cfg80211.h
index f008298332a6..c7c8e047a845 100644
--- a/include/net/cfg80211.h
+++ b/include/net/cfg80211.h
@@ -703,7 +703,6 @@ static inline void wiphy_read_of_freq_limits(struct wiphy *wiphy)
  *	MU-MIMO packets going to the specified station; %NULL if not changed
  * @radio_iface: Radio iface name
  * @mld_macaddr: MLO address to use for this virtual interface.
- * @mld_iface_name: MLO interface name to use for this virtual interface.
  */
 struct vif_params {
 	u32 flags;
@@ -713,7 +712,6 @@ struct vif_params {
 	const u8 *vht_mumimo_follow_addr;
 	char *radio_iface;
 	u8 mld_macaddr[ETH_ALEN];
-	char *mld_iface_name;
 };
 
 /**
diff --git a/include/uapi/linux/nl80211.h b/include/uapi/linux/nl80211.h
index 450bbf73d4b8..a3038578977a 100644
--- a/include/uapi/linux/nl80211.h
+++ b/include/uapi/linux/nl80211.h
@@ -3358,7 +3358,6 @@ enum nl80211_attrs {
 	NL80211_ATTR_MLO_LINK_DISABLED,
 
 	NL80211_ATTR_RADIO_IFACE = 350,
-	NL80211_ATTR_MLD_IFACE_NAME = 351,
 
 	/* add attributes here, update the policy in nl80211.c */
 
diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index cefad1b9cdfc..1902bb63220b 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -819,7 +819,6 @@ static const struct nla_policy nl80211_policy[NUM_NL80211_ATTR] = {
 	[NL80211_ATTR_EMA_RNR_ELEMS] = { .type = NLA_NESTED },
 	[NL80211_ATTR_MLO_LINK_DISABLED] = { .type = NLA_FLAG },
 	[NL80211_ATTR_RADIO_IFACE] = { .type = NLA_BINARY, .len = IFNAMSIZ-1 },
-	[NL80211_ATTR_MLD_IFACE_NAME] = { .type = NLA_BINARY, .len = IFNAMSIZ-1 },
 };
 
 /* policy for the key attributes */
@@ -4297,14 +4296,6 @@ static int _nl80211_new_interface(struct sk_buff *skb, struct genl_info *info)
 			return -ENOTSUPP;
 	}
 
-	if (info->attrs[NL80211_ATTR_MLD_IFACE_NAME]) {
-		if (rdev->wiphy.flags & WIPHY_FLAG_SUPPORTS_MLO)
-			params.mld_iface_name =
-				nla_data(info->attrs[NL80211_ATTR_MLD_IFACE_NAME]);
-		else
-			return -ENOTSUPP;
-	}
-
 	msg = nlmsg_new(NLMSG_DEFAULT_SIZE, GFP_KERNEL);
 	if (!msg)
 		return -ENOMEM;
-- 
2.34.1

